tests/e2e/s3/utils.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python
+
+import subprocess
+import json
+import os
+import sys
+import boto3
+import time
+
+
+def update_agent_biling_clients(billing):
```

Comment:
Could use a comment that answers the following:

1. What does this function do?
2. What is `billing`?
3. What communication is made in order to accomplish this?